Code/Framework/AzToolsFramework/Tests/Prefab/PrefabTestFixture.cpp

```diff
@@ -80,107 +92,123 @@ namespace UnitTest
+        CreatePrefabResult createResult = m_prefabPublicInterface->CreatePrefabInMemory(entityIds, filePath);
+        AZ_Assert(createResult.IsSuccess(), "CreateEditorPrefab - Failed to create prefab %s. Error: %s",
+            AZStd::string(filePath.Native()).c_str(),
+            createResult.GetError().c_str());
 
-        // Update our undo cache entry to include the rename / reparent as one atomic operation.
-        m_prefabPublicInterface->GenerateUndoNodesForEntityChangeAndUpdateCache(entityId, m_undoStack->GetTop());
-        m_prefabSystemComponent->OnSystemTick();
+        // Verify new container entity.
+        AZ::EntityId prefabContainerId = createResult.GetValue();
+        EXPECT_TRUE(prefabContainerId.IsValid());
 
-        return entityId;
+        AZ::Entity* prefabContainerEntity = AzToolsFramework::GetEntityById(prefabContainerId);
+        EXPECT_TRUE(prefabContainerEntity != nullptr);
+
+        PropagateAllTemplateChanges();
+
+        return prefabContainerId;
     }
 
-    AZ::EntityId PrefabTestFixture::CreatePrefab(const AzToolsFramework::EntityIdList& entityIds, AZ::IO::PathView filePath)
+    AZ::EntityId PrefabTestFixture::InstantiateEditorPrefab(AZ::IO::PathView filePath, AZ::EntityId parentId)
     {
-        CreatePrefabResult createPrefabResult = m_prefabPublicInterface->CreatePrefabInMemory(entityIds, filePath);
+        InstantiatePrefabResult instantiateResult = m_prefabPublicInterface->InstantiatePrefab(
+            filePath.Native(), parentId, AZ::Vector3());
+        AZ_Assert(instantiateResult.IsSuccess(), "InstantiateEditorPrefab - Failed to instantiate prefab %s. Error: %s",
+            AZStd::string(filePath.Native()).c_str(),
+            instantiateResult.GetError().c_str());
 
-        // Verify that a valid prefab container entity is created.
-        AZ::EntityId prefabContainerId = createPrefabResult.GetValue();
-        AZ_Assert(prefabContainerId.IsValid(), "CreatePrefab operation resulted in an invalid container entity id.");
-        AZ::Entity* prefabContainerEntity = AzToolsFramework::GetEntityById(prefabContainerId);
-        AZ_Assert(prefabContainerEntity != nullptr, "ContainerEntity created with CreatePrefab() is a nullptr.");
+        // Verify new container entity.
+        AZ::EntityId prefabContainerId = instantiateResult.GetValue();
+        EXPECT_TRUE(prefabContainerId.IsValid());
 
-        // PrefabTestFixture won't add required editor components by default. Hence we add them here.
-        AddRequiredEditorComponents(prefabContainerEntity);
+        AZ::Entity* prefabContainerEntity = AzToolsFramework::GetEntityById(prefabContainerId);
+        EXPECT_TRUE(prefabContainerEntity != nullptr);
+        
+        PropagateAllTemplateChanges();
 
         return prefabContainerId;
     }
 
+    AZ::Entity* PrefabTestFixture::CreateEntity(const AZStd::string& entityName, bool shouldActivate)
```

Comment:
suggestion: A comment mentioning how this is not an editor entity or is specifically a runtime entity might be useful?

Reply:
I will add more comments in header file.

> (also consider switching the bool param to something more readable such as an enum)

I prefer keeping this boolean for now. I may foresee that we should not use this function anymore. imo Since prefab is an editor concept, we should test things within an editor environment. If some old or new tests rely on PrefabTestFixture, I think they should start using the new CreateEditorEntity function.

But I may miss some other test cases that still use CreateEntity or aznew Entity. 🤔